unstract/connectors/src/unstract/connectors/filesystems/google_drive/google_drive.py

```diff
@@ -92,9 +92,16 @@ def get_fsspec_fs(self) -> GDriveFileSystem:
     def test_credentials(self) -> bool:
         """To test credentials for Google Drive."""
         try:
-            self.get_fsspec_fs().isdir("root")
+            is_dir = bool(self.get_fsspec_fs().isdir("root"))
+            if not is_dir:
+                raise RuntimeError(
+                    "Could not connect to its root directory 'root', "
```

Comment:
@chandrasekharan-zipstack In cases of error, should we actually expose that we are trying to connect to the `root` directory in our test connection? Can the error message be more generic ?

Reply:
I had a similar doubt as well since its internal to the connector itself. You're right - it won't make sense to the user and doesn't help debug the issue better. Will refrain from mentioning it

Reply:
Also I'm not sure if this particular message will ever get thrown to the user - I just added it as a fail safe. Usually connection failures get raised as exceptions from the connector clients

Reply:
In that case, if it improves developer debugging experience, can we log it?